packages/frontend/src/components/MetricQueryData/DrillDownModal.tsx

```diff
@@ -203,43 +201,40 @@ const DrillDownModal: FC = () => {
     }, [closeDrillDownModal]);
 
     return (
-        <Dialog
-            isOpen={isDrillDownModalOpen}
+        <Modal
+            opened={isDrillDownModalOpen}
             onClose={closeDrillDownModal}
-            lazy
-            title={`Drill into "${value}"`}
+            title={<Title order={4}>{`Drill into "${value}"`}</Title>}
         >
-            <form>
-                <div className={Classes.DIALOG_BODY}>
-                    <FormGroup
-                        label="Pick a dimension to segment your metric by"
-                        labelFor="chart-name"
+            <Stack>
+                <FieldSelect
+                    withinPortal
+                    disabled={dimensionsAvailable.length === 0}
+                    item={selectedDimension}
+                    items={dimensionsAvailable}
+                    onChange={setSelectedDimension}
+                />
+                <Group position="right">
+                    <Button variant="outline" onClick={onClose}>
+                        Cancel
+                    </Button>
+                    <Button
+                        component="a"
+                        target="_blank"
+                        href={url}
+                        leftIcon={<MantineIcon icon={IconExternalLink} />}
+                        disabled={!selectedDimension}
+                        // TODO: fix this timeout
+                        onMouseUpCapture={() =>
+                            setTimeout(() => {
+                                onClose();
+                            }, 500)
+                        }
```

Comment:
@IrakliJani , need your help with this. 
I want to:
* clear the `selectedDimension` and close the modal on click
* but if I don't add a timeout, it will close the modal + clear the dimension but the link won't open on a new tab. What can be done here? 🤔